### ISSUE TITLE
Elastic: UI blocked during refresh

### ISSUE THREAD

**Issue by manfredu** [roundcube/roundcubemail#7164]
I do like the new elastic theme but it does block the ui while refreshing and I do have to wait quit often before I can continue with working on my mails. 

I have many message folders with messages from 20 years, maybe this is the issue, but still, this does not happen with larry...

**Comment by alecpl** [comment]
It is blocking in other skins too. It is only not blocking mail preview/open action. I don't remember if it ever was non-blocking, maybe it was. We'll need to investigate if it's possible to make the refresh non-blocking.

**Comment by manfredu** [comment]
You are right, other skins block the ui as well (except mail opening, as you said) and in my previous installation it was blocking as well. Only I didn't notice.

I recently switched from 1.1.0 Larry to 1.4.1 Elastic, the difference is that with 1.1.0 the refresh lasts only about 1 seconds whereas with 1.4.1 it takes about 10 seconds. This is using the same imap server. On 1.4.1 Larry the refresh is slow as well so it is not the new Elastic skin.

**Comment by manfredu** [comment]
I've disabled all plugins temporarily, but the refresh is still taking 20 seconds, while with the old installation (which is still available) it is instantaneously.  :-(

**Comment by alecpl** [comment]
Enable imap_debug and compare IMAP commands and timing of these.

**Comment by manfredu** [comment]
Both logs show "OK Search completed (0.000 secs)." for the individual imap folders. But the old installation takes two seconds overall to scan all 285 mailbox folders, whereas the new installation needs 25 seconds. One explanation would be network overhead.  The old system resides on the same machine as the imap server whereas the new one uses a vpn connection to the imap server.

So it would really help me, if the refresh could work in the background without blocking or at least allows switching of folders.

Or, is there a way to configure only individual folders to be checked for new messages? 

I have server-side filters installed moving incoming messages to other folders, therefore I have "Check all folders for new messages" enabled. But these are only a few important folders, it is not necessary that all 285 folders are checked periodically.

**Comment by robsheldon** [comment]
Funny, I've just bumped into this issue under similar circumstances. I previously had a roundcube instance running on the mail server; as web hosting requirements have changed over the years, it makes more sense now to run roundcube on the web server and have it reach out to the mail server.

In that setup, the "Refreshing..." UI issue becomes a constant nuisance. I find myself trying to rush to perform actions in between "Refreshing..." popups.

For dovecot environments, it may make sense to use dsync to create a local imap mirror (I haven't tried this yet).

**Comment by manfredu** [comment]
I helped myself by disabling the automatic refresh of all folders. Now I start the complete refresh manually by clicking on the button, whenever this is convenient for me.

Nevertheless I still get the 25 seconds break after sending a mail. For whatever reason Roundcube always does a complete refresh afterwards.

**Comment by manfredu** [comment]
> Nevertheless I still get the 25 seconds break after sending a mail. For whatever reason Roundcube always does a complete refresh afterwards.

Is there a way to get around this?

**Comment by mckaygerhard** [comment]
i have problem with newer versions of roundcube.. with inbox more than 4000+ mails.. but that problem does not happened with older 0.9.5 version.. see #7300 seems newer version of roundcube has performance issues.. very !

**Comment by alecpl** [comment]
It could be fixed/mitigated with:
1. not blocking the UI while refreshing.
2. using LIST-STATUS extension - RFC 5819.

**Comment by crzidea** [comment]
> 2. using LIST-STATUS extension - RFC 5819.

Is there any update for this?
